view/authentication/login/app.js

```diff
@@ -35,6 +35,23 @@ function getQueryParams(qs) {
   return params;
 }
 
+/**
+ * Check if the input string is a valid URL of protocol http or https
+ * This should invalidate the XSS URL redirect of protocal 'javascript:'
+ * @param {String} url url string to be validated
+ * @returns {Boolean} true if the URL is valid
+ */
+function isValidURL(string) {
+  const decodedString = decodeURIComponent(string);
+  let url;
+  try {
+    url = new URL(decodedString);
+  } catch (_) {
+    return false;
+  }
+  return url.protocol === 'http:' || url.protocol === 'https:';
```

Comment:
This is a *very* small thing and not a blocker at all but just a super-tiny-micro-performance-enhancement: think about how this is going to be called the most... it will almost *always* be called with url.protocol === "https:" so that should be the first check on line 52 rather than the 2nd.

Reply:
also: add some unit tests for the isValidUrl fn with some different strings?